### DISCUSSION TITLE
Not able to create workouts

### DISCUSSION THREAD

**Discussion by ckelwin** [EXERLOG/exer_log#90] [General]
Is it just me? Things appear to be slower than previously. And after creating my workouts and going back to the calendar screen, the workout doesn't appear on the day. I'm now on the `alpha` branch but I've also tried a few other branches.

Console shows the message below.
`W/Firestore( 4229): (24.2.1) [WriteStream]: (85f7abd) Stream closed with status: Status{code=UNAVAILABLE, description=Channel shutdownNow invoked, cause=null}.`

**Comment by momshaddinury** [comment]
I just worked on #93 PR. I was able to create a workout. But yeah, someone also said something like this somewhere. Their issue was bad state. Though while creating workout I am getting a hero tag duplicate issue.

**Reply by thefr0gg** [reply]
Yeah it was me i found a fix for it and that is by changing a line in this function (this is the file for it [max_bloc.dart](https://github.com/KalleHallden/exer_log/blob/alpha/lib/Bloc/max_bloc.dart)) 
``` dart
Future<Max> getOneRepMax(String exercise) async {
  int reps = 1;
  QuerySnapshot<Map<String, dynamic>>? ref;
  ref = await FirebaseFirestore.instance
        .collection('users')
        .doc(userID)
        .collection('maxes')
        .where('exercise', isEqualTo: exercise)
        .orderBy('weight', descending: true)
        .get();
  print(ref.docs.first.data());
  Max returnMax;
  if (ref != null) {
    returnMax = Max.fromJson(ref.docs.first.data());
  } else {
    returnMax = new Max(0, 0, 0, '');
  }
  return returnMax;
}
```
check if the ```ref``` if it empty and not if it null like this
``` dart
 if (ref.docs.isNotEmpty) {
    returnMax = Max.fromJson(ref.docs.first.data());
  } else {
    returnMax = new Max(0, 0, 0, '');
  }
```

**Reply by momshaddinury** [reply]
@0RaMsY0 when you say it was you, do you mean you created this bug? Or you figured out the issue?

**Reply by thefr0gg** [reply]
I figured out the issue and i try to look for how to solve it 🙂 
PR #96 that will fix the issue

**Comment by thefr0gg** [comment]
can  i ask why aren't passing here the ```setMax``` parameter into ```MaxInformation```
![image](https://user-images.githubusercontent.com/86024158/185806908-0fbfbc09-87be-4c02-9343-65b13a302c04.png)
because this well raise an error when trying to view the workout that i made previously by clicking on the day number from the calender.
![IMG_20220821_200734](https://user-images.githubusercontent.com/86024158/185807020-2c382c50-b6d6-4874-ad48-f5624ac3df31.png)

**Reply by momshaddinury** [reply]
The overflow issue is not fixed yet? Wasn't there a PR?

**Reply by thefr0gg** [reply]
@dinurymomshad it didn't get merged
And i think the overflow is happening because the days number are buttons and they won't fit in the Container so we have to add a little bit of height into it

**Reply by momshaddinury** [reply]
> @dinurymomshad it didn't get merged
> And i think the overflow is happening because the days number are buttons and they won't fit in the Container so we have to add a little bit of height into it

We don't have to add any fixed weight. We just need to make the container hight dynamic. I'll look into it.

**Reply by thefr0gg** [reply]
@dinurymomshad alright

**Comment by ckelwin** [comment]
Attaching a screen recording for reference. Still not able to create workouts and it's blocking me.

https://user-images.githubusercontent.com/5429312/187013246-05e7511c-6586-4d87-b642-8dadf35b5f31.mp4

**Comment by ckelwin** [comment]
Just an update. I was having the said problem with creating and/or viewing workouts when using an emulator. Switched to a physical device and it's working fine now. Much faster as well.

**Reply by momshaddinury** [reply]
Can you check your emulator wifi connectivity? Sometimes emulator connectivity bugs out.
